### DISCUSSION TITLE
Win 32bits support

### DISCUSSION THREAD

**Discussion by sanjuruk** [bluenviron/mediamtx#264] [Questions]
## Which version are you using?

v0.12.0

## Which operating system are you using?

Windows

## Question

I see on release page, assets only support Windows64 bits. 

1. Can I find a build for 32bits in Windows?
2. If not, can I compile the binary to make an .exe file to support Windows 32bits? If yes, how?

Can someone point me at the right direction?

Thanks.

**Comment by aler9** [comment]
Hello, builds for Windows 32 are currently not provided, since the platform is seldom used, and since supporting it would probably lead to the discovery of 32bit-only bugs (usually caused by int overflows and wrong structs alignment) and a lot of effort to fix them.

If more people ask for win32 support, i'll add it to the list of supported platforms.

You can cross-compile the software for win32 from a Linux platform in this way:
```
GOOS=windows GOARCH=386 go build -o rtsp-simple-server.exe .
```

**Comment by sanjuruk** [comment]
Thank you for the reply. I successfully compiled for 32bits with the above command. However, when the exe is run on 32bit XP, I get an error "not a valid Win32 application"  

I referred the [article on CrossCompilation](https://github.com/golang/go/wiki/WindowsCrossCompiling). Still same error occurs even on small hello.go application. 
Probably some compilation issue. Have you succeeded on running simple go applications on Win32?

**Comment by aler9** [comment]
Unfortunately Windows XP was last supported in Go 1.10, which is not compatible with rtsp-simple-server.
To compile rtsp-simple-server, the minimum Go version is 1.15, hence the oldest Windows version for cross-compilation is Windows 7 32 bit.

**Comment by sanjuruk** [comment]
Thanks for the clarification 👍

**Comment by sanjuruk** [comment]
Does that mean the [windows_amd64](https://github.com/aler9/rtsp-simple-server/releases/download/v0.12.0/rtsp-simple-server_v0.12.0_windows_amd64.zip) exe only supports Windows 7 and later 64bits systems?

**Comment by aler9** [comment]
yes, OS requirements are dictated by Go, but this means supporting almost all modern platforms and OSs (including all Linux distros, since binaries don't depend on libc).

If you need a machine to host a server, the advice i always give is go buy a Raspberry Pi (35$) or either a VPS (40$/year).

[a comment by an automated account is not shown]
